app/models/pipeline_run.rb

```diff
@@ -853,8 +843,11 @@ def update_job_status
       if prs.failed?
         self.job_status = STATUS_FAILED
         self.finalized = 1
-        message = "SampleFailedEvent: Sample #{sample.id} by #{sample.user.email} failed #{prs.name} with #{adjusted_remaining_reads || 0} reads remaining after #{duration_hrs} hours. See: #{status_url}"
-        LogUtil.log_err_and_airbrake(message)
+        self.known_user_error, self.error_message = check_for_user_error(prs)
+        log_message = "SampleFailedEvent: Sample #{sample.id} by #{sample.user.admin? ? 'admin user' : 'non-admin user'} " \
+          "failed #{prs.name} with #{adjusted_remaining_reads || 0} reads remaining after #{duration_hrs} hours. " \
+          "See: #{status_url}"
+        LogUtil.log_err_and_airbrake(log_message) unless known_user_error
```

Comment:
we should count the `known_user_error` somehow even if not showing as a pipeline failure

Reply:
Good idea, how/where would you like to be alerted to this? My thinking was that we're trying to denoise alerts. Anyone who is interested could still count known_user_errors in Mode or see them in the sample table as "complete - issue"

Reply:
I’m coding the restart logic of our watchdog and I’m about to touch this piece of code. I’m planning to move this message to a different place, which aligns to Charles’ idea of “denoising” alerts (I can carry Greg’s logic with my change).
I’m planning to just log an error message here without sending it to air brake, and enqueue in resque an asynchronous pipeline restart in case this is the first error for this sample. I’ll only trigger air brake if this is not the first failure.